src/main/java/core/basesyntax/FileWriterImpl.java

```
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import core.interfaces.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+public class FileWriterImpl implements FileWriter {
+
+    @Override
+    public void write(String report, String reportName) {
+        Path path = Path.of(reportName);
+
+        try {
+            if (!Files.exists(path)) {
+                Files.createFile(path);
+            }
+
+            Files.writeString(
+                    path,
+                    report,
+                    StandardOpenOption.TRUNCATE_EXISTING,
+                    StandardOpenOption.CREATE);
+        } catch (IOException e) {
+            throw new RuntimeException(e);

```

Catching IOException and rethrowing `new RuntimeException(e)` loses context. The checklist explicitly says: "Don't ignore exceptions in try-catch blocks ... Let's rethrow a RuntimeException with an **informative** message and exception object." Provide an informative message that includes the file path, for example: `throw new RuntimeException("Can't write file by path: " + reportName, e);`